test/uk/gov/hmrc/helptosave/connectors/HelpToSaveProxyConnectorSpec.scala

```diff
@@ -293,17 +295,46 @@ class HelpToSaveProxyConnectorSpec extends TestSupport with MockPagerDuty with E
         mockGetAccountResponse(getAccountUrl)(Some(HttpResponse(400)))
         mockPagerDutyAlert("Received unexpected http status in response to getAccount")
 
-        val result = await(proxyConnector.getAccount(nino, queryString).value)
+        val result = await(proxyConnector.getAccount(nino, systemId, correlationId).value)
         result shouldBe Left("Received unexpected status(400) from getNsiAccount call")
       }
 
       "handle unexpected server errors" in {
         mockGetAccountResponse(getAccountUrl)(None)
         mockPagerDutyAlert("Failed to make call to getAccount")
 
-        val result = await(proxyConnector.getAccount(nino, queryString).value)
+        val result = await(proxyConnector.getAccount(nino, systemId, correlationId).value)
         result.isLeft shouldBe true
       }
+
+      "handle the case where an account does not exist" in {
+        val errorResponse =
+          Json.parse(
+            s"""
+              |{
+              |  "errors" : [
+              |    {
+              |      "errorMessageId" : "id",
+              |      "errorMessage"   : "message",
+              |      "errorDetail"    : "detail"
+              |    },
+              |    {
+              |      "errorMessageId" : "${appConfig.runModeConfiguration.underlying.getString("nsi.get-account.no-account-error-message-id")}",
+              |      "errorMessage"   : "Oh no!",
+              |      "errorDetail"    : "Account doesn't exist"
+              |    }
+              |  ]
+              |}
+          """.stripMargin)
+
+        mockGetAccountResponse(getAccountUrl)(Some(HttpResponse(400, Some(errorResponse))))
+        mockPagerDutyAlert("Received unexpected http status in response to getAccount")
```

Comment:
I'm not sure whether you'd want an alert in this case, it could be normal for the specified NINO to not have a Help to Save account (maybe a caller might use this endpoint to find out whether an account exists for the specified NINO).

In the short term, given the way the mobile services call it I wouldn't expect this alert to be fired so you could leave it in until/unless you notice a problem with false alarms.

Another way of dealing with it could be to check enrolments in the `help-to-save` mongo database before calling NS&I. That way it _would_ be an error that you'd want to be alerted about if a user who was enrolled our side didn't have an account on the NS&I side. Not something that has to be done in this story though, just an idea.

Reply:
I might make sense to check our enrolment store before the call to NS&I - we can save an outbound call to NS&I if we see they don't have an account. The only case where this might be a problem is if the person has created an account but there has been a problem writing a record in our enrolment store for that user. Then if we're checking our enrolment store in this endpoint first, for this person we're blocking the call where it would have otherwise been successful. I think then, if we have the current behaviour where we don't check our enrolment store, then you're right - a pager duty alert is not needed in this case since it is kind of "expected" behaviour